### ISSUE TITLE
[Python] Remove `use_legacy_dataset` from code base

### ISSUE THREAD

**Issue by raulcd** [apache/arrow#44790]
### Describe the enhancement requested

We have been warning for a while about legacy dataset being deprecated since Arrow 15.0.0.
Should we remove the warnings and deprecate the `use_legacy_dataset`  keyword arguments?

Legacy dataset was removed here (from what I can find):
- https://github.com/apache/arrow/issues/31303

I haven't been able to find an issue to remove that, that's why I am opening it.

### Component(s)

Python

**Comment by raulcd** [comment]
@jorisvandenbossche @AlenkaF is this something we might potentially want to do now? I might be able to try and find some time to do it.

**Comment by raulcd** [comment]
cc @pitrou

**Comment by AlenkaF** [comment]
I do not think we created an issue for that looking at the discussion around the `use_legacy_dataset` keyword arguments under this part of the review: https://github.com/apache/arrow/pull/39112#pullrequestreview-1769501660

so thanks for opening one!
I think we could do this now, it has been a year since the keyword has been deprecated.

**Comment by AlenkaF** [comment]
Issue resolved by pull request 45742
https://github.com/apache/arrow/pull/45742
